examples/rdkafka_performance.c

```diff
@@ -962,37 +962,41 @@ int main (int argc, char **argv) {
 		case 'd':
 			debug = optarg;
 			break;
-                case 'H':
-                {
-                        char *name, *val;
-                        size_t name_sz = -1;
-
-                        if (!optarg) {
-                                read_hdrs = 1;
-                                break;
-                        }
-
-                        name = optarg;
-                        val = strchr(name, '=');
-                        if (val) {
-                                name_sz = (size_t)(val-name);
-                                val++; /* past the '=' */
-                        }
+        case 'h':
+            read_hdrs = 1;
+            break;
+        case 'H':
+            if (mode == 'P') {
```

Comment:
This requires the mode to be specified prior to headers, which is not required by other options.

Reply:
Suggest just checking for "parse" and then later on after the argument check is complete verify that mode is consumer/group-consumer when read_hdrs are set.

Reply:
ok, will check after argument parsing but before client creation